### DISCUSSION TITLE
Home Assistant Energy dashboard integration

### DISCUSSION THREAD

**Discussion by JakobTewes** [syssi/esphome-jk-bms#370] [Q&A] [answered]
Heyhey,

wanted to add my batteries to the HA Energy Dashboard. For successful integration, I´d need "Energy going to the battery" and "Energy coming out of the battery". The element is described as "Pick sensors which measure energy going in to and out of the battery in either of GJ, kWh, MJ, MWh, Wh."

Someone already integrated their storage to the dashboard? Do you have an idea, what (template sensors) I need to create, in order to get the storage integrated?

Thanks and regards

Jakob

**Comment by syssi** [comment, marked as answer]
Did you see this example? https://github.com/syssi/esphome-jk-bms/blob/main/yaml-snippets/esp32-ble-energy-dashboard.yaml

**Reply by JakobTewes** [reply]
Nice, @syssi! You’re so fast and helpful - thanks so much (again and again ;-)).

Will give the thing a try and ping back then 👍

**Reply by JakobTewes** [reply]
Perfect - works like a charm!

**Reply by TheOriginal92** [reply]
@syssi sorry to open this up again, is there a example configuration for the energy dashboard using uart/modbus instead of BLE? Thank you!
